Update the agile development contents in sprint planning and backlog refinement. (#423)
microsoft/code-with-engineering-playbook: agile-development/backlog-management/refinement/readme.md

What are the intended outcomes of the ceremonies?

> Backlog refinement or formerly known as [backlog grooming](https://www.agilealliance.org/glossary/backlog-grooming/) is when the [product owner](https://www.agilealliance.org/glossary/product-owner/) and some, or all, of the rest of the team review items on the backlog to ensure the backlog contains the appropriate items, that they are prioritized, and that the items at the top of the backlog are ready for delivery. This activity occurs on a regular basis and may be an officially scheduled meeting or an ongoing activity.

Reviewer: what do you think about ready for development over ready for delivery? either way is good, just thinking they have a bit different meanings

> Backlog refinement or formerly known as [backlog grooming](https://www.agilealliance.org/glossary/backlog-grooming/) is when the [product owner](https://www.agilealliance.org/glossary/product-owner/) and some, or all, of the rest of the team review items on the backlog to ensure the backlog contains the appropriate items, that they are prioritized, and that the items at the top of the backlog are ready for delivery. This activity occurs on a regular basis and may be an officially scheduled meeting or an ongoing activity.

Reviewer: ```suggestion
Backlog refinement or formerly known as [backlog grooming](https://www.agilealliance.org/glossary/backlog-grooming/) is when the [product owner](https://www.agilealliance.org/glossary/product-owner/) and some, or all, of the rest of the team review items on the backlog to ensure the backlog contains the appropriate items, that they are prioritized, and that the items at the top of the backlog are ready for development. This activity occurs on a regular basis and may be an officially scheduled meeting or an ongoing activity.
```

## Participation

What members of the team should participate?

Ideally all of the team members could participate. This is an ongoing process in which the Product Owner and the Development Team collaborate on the details of Product Backlog items.

## Impact

What positive impact has been observed as a result of this practice?

> The intent of backlog refinement is to ensure that the backlog keeps updating with items that are appropriate to their priority, i.e. higher priority items are with more details than lower priority ones. Also it is important that at any time there are enough user stories ready for planning in the next or even, where reasonable, the next two or three iterations but not too further away.

Reviewer: nit: not too further away => not too much further

> The intent of backlog refinement is to ensure that the backlog keeps updating with items that are appropriate to their priority, i.e. higher priority items are with more details than lower priority ones. Also it is important that at any time there are enough user stories ready for planning in the next or even, where reasonable, the next two or three iterations but not too further away.

Reviewer: ```suggestion
The intent of backlog refinement is to ensure that the backlog keeps updating with items that are appropriate to their priority, i.e. higher priority items are with more details than lower priority ones. Also it is important that at any time there are enough user stories ready for planning in the next or even, where reasonable, the next two or three iterations but not too much further.
```

## Measures

How might one measure the impact of this ceremony positive or negative?

> The expected outcome from the backlog refinement is all team members are on the same page of what they are going to implement in the next or next few iterations with priority. A positive outcome of refinement is one where each backlog item can reasonably be “Done” within the Sprint time - box. Product Backlog items that can be “Done” by the Development Team within one Sprint are deemed “Ready” for selection in a Sprint Planning.

Reviewer: nit: is all team members => is that all team members

> The expected outcome from the backlog refinement is all team members are on the same page of what they are going to implement in the next or next few iterations with priority. A positive outcome of refinement is one where each backlog item can reasonably be “Done” within the Sprint time - box. Product Backlog items that can be “Done” by the Development Team within one Sprint are deemed “Ready” for selection in a Sprint Planning.

Reviewer: nit: Sprint time - box => sprint time

> The expected outcome from the backlog refinement is all team members are on the same page of what they are going to implement in the next or next few iterations with priority. A positive outcome of refinement is one where each backlog item can reasonably be “Done” within the Sprint time - box. Product Backlog items that can be “Done” by the Development Team within one Sprint are deemed “Ready” for selection in a Sprint Planning.

Reviewer: nit: I think we should go for all lowercase within a sentence - here we have Sprint Planning, Done, Product Backlog item, Development Team, Ready - is there a reason why these are title cased inside the sentence?

If you are looking for emphasis - use **bold** or `tics` 

This is throughout the document

> The expected outcome from the backlog refinement is all team members are on the same page of what they are going to implement in the next or next few iterations with priority. A positive outcome of refinement is one where each backlog item can reasonably be “Done” within the Sprint time - box. Product Backlog items that can be “Done” by the Development Team within one Sprint are deemed “Ready” for selection in a Sprint Planning.

Reviewer: ```suggestion
The expected outcome from the backlog refinement is all the team members are on the same page of what they are going to implement in the next or next few iterations with priority. A positive outcome of refinement is one where each backlog item can reasonably be “done” within the sprint time. Product backlog items that can be “done” by the development team within one sprint are deemed “ready” for selection in a sprint planning.
```

## Facilitation Guidance

How might one go about running the ceremony? Is there a general pattern that the ceremony should follow?

- Product Owner: presents the ordered backlog and backlog list they'd like in the next iteration.
- Process Lead: ensure the timebox.

> - Team members: asks questions and requests additional details on the backlog if needed.

Reviewer: nit: asks => ask and requests => request (as team members are plural)

> - Team members: asks questions and requests additional details on the backlog if needed.

Reviewer: ```suggestion
- Team members: ask questions and request additional details on the backlog if needed.
```

What are the intended outcomes of the ceremonies?

Backlog refinement ( or formerly known as [backlog grooming](https: // www.agilealliance.org/glossary/backlog-grooming /) is when the [product owner](https: // www.agilealliance.org/glossary/product-owner /) and some, or all, of the rest of the team review items on the backlog to ensure the backlog contains the appropriate items, that they are prioritized, and that the items at the top of the backlog are ready for delivery. This activity occurs on a regular basis and may be an officially scheduled meeting or an ongoing activity.

## Participation

What members of the team should participate?

Ideally all of the team members could participate. This is an ongoing process in which the Product Owner and the Development Team collaborate on the details of Product Backlog items.

## Impact

What positive impact has been observed as a result of this practice?

> The intent of backlog refinement is to ensure that the backlog keeps updating with items that are appropriate to their priority, i.e. higher priority iteams are with more details than lower priority ones . Also its important that at any time there are enough user stories ready for planning in the next or even, where reasonable, the next two or three iterations but not too further away.

Reviewer 2: *nit*: spacing between `ones` and `.`

## Measures

How might one measure the impact of this ceremony positive or negative?

The expected outcome from the backlog refinement is all team members are on the same page of what they are going to implement in the next or next few iterations with priority. A positive outcome of refinement is one where each backlog item can reasonably be “Done” within the Sprint time - box. Product Backlog items that can be “Done” by the Development Team within

> one Sprint are deemed “Ready” for selection in a Sprint Planning.

Reviewer 2: *nit*: unnecessary enter
